ArduPlane/commands_logic.cpp

```diff
@@ -383,6 +390,58 @@ void Plane::do_nav_wp(const AP_Mission::Mission_Command& cmd)
     set_next_WP(cmd.content.location);
 }
 
+inline int modulo(int a, int b) {
+    if (b < 0) {
+        return modulo(-a, -b);
+    }
+    const int result = a % b;
+    return result >= 0 ? result : result + b;
+}
+
+void Plane::do_nav_into_wind(const AP_Mission::Mission_Command& cmd)
+{
+    Vector3f wind = ahrs.wind_estimate();
+    float wind_to = M_PI_2 - atan2f(wind.y, wind.x);
+    if (wind_to < 0.0) {
+        wind_to += M_2PI;
+    }
+
+    //get the next waypoint position
+    AP_Mission::Mission_Command next_cmd;
+    const uint16_t current_index = mission.get_current_nav_index() + 1;
+    mission.get_next_nav_cmd(current_index, next_cmd);
```

Comment:
This can return false if there isn't a next nav cmd. That needs to be explicitly handled, I'd say that in this case just considering the mission complete would be fine.